Source/SwiftLintBuiltInRules/Rules/Lint/UnneededOverrideRule.swift

```diff
@@ -16,103 +16,136 @@ struct UnneededOverrideRule: Rule {
+    var modifiers: DeclModifierListSyntax { get }
+    var attributes: AttributeListSyntax { get }
+    var signature: FunctionSignatureSyntax { get }
+    var body: CodeBlockSyntax? { get }
+    var declText: String { get }
 }
 
-private extension UnneededOverrideRule {
-    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
-        override func visitPost(_ node: FunctionDeclSyntax) {
-            if isUnneededOverride(node) {
-                self.violations.append(node.positionAfterSkippingLeadingTrivia)
-            }
+private extension OverridableDeclSyntax {
+    var isUnneededOverride: Bool {
+        guard modifiers.contains(keyword: .override), let statement = body?.statements.onlyElement else {
+            return false
         }
-    }
 
-    final class Rewriter: ViolationsSyntaxRewriter {
-        override func visit(_ node: FunctionDeclSyntax) -> DeclSyntax {
-            if isUnneededOverride(node) {
-                correctionPositions.append(node.positionAfterSkippingLeadingTrivia)
-                let expr: DeclSyntax = ""
-                return expr
-                    .with(\.leadingTrivia, node.leadingTrivia)
-                    .with(\.trailingTrivia, node.trailingTrivia)
-            }
+        // Assume having @available changes behavior.
+        if attributes.contains(attributeNamed: "available") {
+            return false
+        }
+
+        guard let call = extractFunctionCallSyntax(statement.item),
+            let member = call.calledExpression.as(MemberAccessExprSyntax.self),
+              member.base?.is(SuperExprSyntax.self) == true,
+              member.declName.baseName.text == declText else {
+            return false
+        }
+
+        let declParameters = signature.parameterClause.parameters
+        if declParameters.contains(where: { $0.defaultValue != nil }) {
+            // Any default parameter might be a change to the super function.
+            return false
+        }
+
+        // Assume any change in arguments passed means behavior was changed.
+        let expectedArguments = declParameters.map {
+            ($0.firstName.text == "_" ? "" : $0.firstName.text, $0.secondName?.text ?? $0.firstName.text)
+        }
+        let actualArguments = call.arguments.map {
+            ($0.label?.text ?? "", $0.expression.as(DeclReferenceExprSyntax.self)?.baseName.text ?? "")
+        }
+
+        guard expectedArguments.count == actualArguments.count else {
+            return false
+        }
 
-            return super.visit(node)
+        for (lhs, rhs) in zip(expectedArguments, actualArguments) where lhs != rhs {
+            return false
         }
+
+        return true
     }
 }
 
-private func isUnneededOverride(_ node: FunctionDeclSyntax) -> Bool {
-    guard node.modifiers.contains(keyword: .override), let statement = node.body?.statements.onlyElement else {
-        return false
+extension FunctionDeclSyntax: OverridableDeclSyntax {
```

Comment:
Please make these two extensions `private` as well.

Reply:
I tried making them `private` while I was implementing this, but we can't:
```
'private' modifier cannot be used with extensions that declare protocol conformances
```

Reply:
I see. Please avoid to add the `declText` to the types and make it visible to the whole module. Rather pass the name to `isUnneededOverride` which you'd need to make a function.

Reply:
Done.